### PR TITLE
[Form] use a reference date to handle times during DST

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #18366
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: UPGRADE-4.4.md [hunk 1]
```diff
@@ -72,6 +72,8 @@ Filesystem
 Form
 ----
 
+ * Using different values for the "model_timezone" and "view_timezone" options without configuring a reference date is
```
Comment (stof) [inline UPGRADE-4.4.md]: you should say `the "model_timezone" and "view_timezone" options of the TimeType`. These options also exist for other types (DateType and DatetimeType) for which there is no such deprecation AFAICT.

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToArrayTransformer.php [hunk 2]
```diff
@@ -41,6 +42,7 @@ public function __construct(string $inputTimezone = null, string $outputTimezone
 
         $this->fields = $fields;
         $this->pad = $pad;
+        $this->referenceDate = $referenceDate ?: new \DateTime('1970-01-01 00:00:00');
```
Comment (stof) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToArrayTransformer.php]: should we use a `DateTimeImmutable` by default ?

**Comment by xabbuh** [comment]
Status: Needs Review

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**